### PR TITLE
feat(cheatcodes): add vm.cloneAccount() cheatcode

### LINKED ISSUE #4893: Add `vm.cloneAccount` cheatcode for cloning account to a specific address
### Component

Forge

### Describe the feature you would like

We are interested in a new cheatcode op: `vm.cloneAccount`, which clones (code, state, balance, nonce) from one account to another account. This would be useful when we need to deploy code with a complex constructor to a specific address, let external calls from a contract we cannot control execute different code on different accounts when we fork the chain, etc. These cannot be easily achieved using `vm.deal`, `vm.etch` or `vm.accesses`

This operation is hard to implement in Solidity but easy in revm. 

Example: 
```
vm.cloneAccount(0x35c9dfd76bf02107ff4f7128Bd69716612d31dDb, 0xeCf9571038d92097a6C09A8d75713fEE69877211)
```

which should clone (code, state, balance, nonce) of 0x35c9dfd76bf02107ff4f7128Bd69716612d31dDb to 0xeCf9571038d92097a6C09A8d75713fEE69877211.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by grandizzy**
## Motivation
Closes #4893 

## Solution

- add `copy_account` to `DatabaseExt` trait and reuse impl in `load_allocs`
- `cloneAccountCall` to copy account and mark target as persistent

### CODE REVIEW

File: testdata/default/cheats/CloneAccount.t.sol [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        b = address(111);
+        c[0] = 222;
+        c[1] = 333;
+        c[2] = 444;
+        d = true;
+    }
+
+    function getC(uint256 index) public returns (uint256 val) {
+        val = c[index];
+    }
```
Comment (DaniPopes) [inline testdata/default/cheats/CloneAccount.t.sol:24]: this is identical to auto-generated `c` getter

File: crates/cheatcodes/spec/src/vm.rs [hunk 2]
```diff
@@ -287,6 +287,10 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Unsafe)]
     function loadAllocs(string calldata pathToAllocsJson) external;
 
+    /// Clones a source account code, state, balance and nonce to a target account and updates in-memory revm state.
```
Comment (DaniPopes) [inline crates/cheatcodes/spec/src/vm.rs:290]:
```suggestion
    /// Clones a source account code, state, balance and nonce to a target account and updates in-memory EVM state.
```

Above too

Comment (grandizzy) [reply crates/cheatcodes/spec/src/vm.rs]: yep, I included changes in https://github.com/foundry-rs/foundry/pull/9048/commits/77e0033c5aab132819b24b179cadc8dc6d4eb865 to save a CI failed run due to cargo cheats

File: crates/evm/core/src/backend/cow.rs [hunk 3]
```diff
@@ -233,6 +233,15 @@ impl DatabaseExt for CowBackend<'_> {
         self.backend_mut(&Env::default()).load_allocs(allocs, journaled_state)
     }
 
+    fn copy_account(
```
[the remark this answers is not in the document]
Comment (grandizzy) [reply crates/evm/core/src/backend/cow.rs]: :+1: done https://github.com/foundry-rs/foundry/pull/9048/commits/77e0033c5aab132819b24b179cadc8dc6d4eb865

File: crates/cheatcodes/src/evm.rs [hunk 4]
```diff
@@ -960,3 +960,20 @@ fn get_state_diff(state: &mut Cheatcodes) -> Result {
         .collect::<Vec<_>>();
     Ok(res.abi_encode())
 }
+
+/// Helper function that creates a `GenesisAccount` from a regular `Account`.
+fn genesis_account(account: &mut Account) -> GenesisAccount {
```
[the remark this answers is not in the document]
Comment (grandizzy) [reply crates/cheatcodes/src/evm.rs]: removed in https://github.com/foundry-rs/foundry/pull/9048/commits/77e0033c5aab132819b24b179cadc8dc6d4eb865

File: testdata/default/cheats/CloneAccount.t.sol (hunk 1 above)
Comment (grandizzy) [reply testdata/default/cheats/CloneAccount.t.sol]: removed in https://github.com/foundry-rs/foundry/pull/9048/commits/77e0033c5aab132819b24b179cadc8dc6d4eb865

**Comment by DaniPopes** [comment]
Does this also address `copyStorage` from https://github.com/foundry-rs/foundry/issues/4072 ?

**Review by DaniPopes — APPROVED** [review]

**Comment by grandizzy** [comment]
> We already have copyStorage, should we go with copy* here as well?

I think I like `clone` here more as it suggest it can have as target an address that doesn't exist yet, while `copyStorage` is more for copying between existing contracts. Happy to change though.

**Comment by DaniPopes** [comment]
That's fine

**Pull request merged**